bin-ranges-transfer/src/main/java/uk/gov/pay/java_lambdas/bin_ranges_transfer/model/Version.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.pay.java_lambdas.bin_ranges_transfer.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+public enum Version {
+    V03, V04, UNKNOWN;
+    
+    private static final List<Version> supportedVersions = Arrays.asList(V03, V04);
+    private static final Logger logger = LoggerFactory.getLogger(Version.class);
+    
+    public static Version fromEnvironment(String versionStr) {
+        for (Version v : supportedVersions) {
+            if (v.name().equals(versionStr.toUpperCase())) {
+                logger.info("Configured Worldpay File Version: {}", v.name());
+                return v;
+            }
+        }
+        throw new IllegalArgumentException("Version not supported: " + versionStr);
+    }
+
+    public static Version fromString(String versionStr) {
+        for (Version v : supportedVersions) {
+            if (v.name().equals(versionStr.toUpperCase())) {
+                logger.debug("Version validated: {}", v.name());
+                return v;
+            }
+        }
+        logger.warn("Version not recognised: {}", versionStr);
+        return UNKNOWN;
```

Comment:
I don’t really understand why we have one method that throws an exception and one that returns `UNKNOWN`.

Reply:
When the lambda starts, we look at the environment to determine the file version we want to use as a candidate. In this case we want the lambda to immediately bail out if the version is not set or is set incorrectly. 

In other cases where we extract a version from the file names on the SFTP server, we don't want to function to error when we encounter a version we don't know about, rather we should warn that this occurred but allow the function to continue.